### PR TITLE
[ALLI-7940] Fix checkout history date handling and Mikromarc function…

### PR DESCRIPTION
**Pull request by EreMaijala**
…ality.

historicloans template was using all-lowercase keys, while upstream template and drivers returned camelCaps. Mikromarc retrieval mechanism was broken. Fixed and added sorting since we always fetch everything from the server.

### CODE REVIEW

File: module/Finna/src/Finna/ILS/Driver/Mikromarc.php [hunk 1]
```diff
@@ -1077,62 +1088,76 @@ public function getDefaultPickUpLocation($patron = false, $holdDetails = null)
...
-            } elseif ($params['sort'] == 'everything desc') {
-                $history['transactions'][] = $transaction;
+            $transactions[$id][$serviceCodeMap[$code]] = $actionTime;
+            // Dates for sorting:
+            $transactions[$id]['_' . $serviceCodeMap[$code]] = $actionTimestamp;
+        }
+
+        // Sort the list:
+        $parts = explode(' ', $params['sort'] ?? '');
+        $sort = $parts[0] ?? 'checkoutDate';
```
Comment (LuomaJuha) [inline module/Finna/src/Finna/ILS/Driver/Mikromarc.php]: Pitäisi defaultata checkout eikä checkoutDate?

File: module/Finna/src/Finna/ILS/Driver/Mikromarc.php [hunk 2]
```diff
@@ -1077,62 +1088,76 @@ public function getDefaultPickUpLocation($patron = false, $holdDetails = null)
...
-                $history['transactions'][] = $transaction;
-            } elseif ($params['sort'] == 'return ' . $sort
-                && isset($transaction['returndate'])
-            ) {
-                $history['transactions'][] = $transaction;
-            } elseif ($params['sort'] == 'everything desc') {
-                $history['transactions'][] = $transaction;
+            $transactions[$id][$serviceCodeMap[$code]] = $actionTime;
+            // Dates for sorting:
+            $transactions[$id]['_' . $serviceCodeMap[$code]] = $actionTimestamp;
```
Comment (LuomaJuha) [inline module/Finna/src/Finna/ILS/Driver/Mikromarc.php]: Vois olla kivempi, jos servicecodemap[code] olis jossain muuttujassa.

**Review by LuomaJuha — CHANGES_REQUESTED** [review]

File: module/Finna/src/Finna/ILS/Driver/Mikromarc.php (hunk 1 above)
Comment (EreMaijala) [reply module/Finna/src/Finna/ILS/Driver/Mikromarc.php]: Juu, korjattu.

File: module/Finna/src/Finna/ILS/Driver/Mikromarc.php (hunk 2 above)
Comment (EreMaijala) [reply module/Finna/src/Finna/ILS/Driver/Mikromarc.php]: Siistitty.

**Review by LuomaJuha — APPROVED** [review]

**Pull request merged**